src/systemcmds/motor_ramp/motor_ramp.cpp

```diff
@@ -317,13 +365,46 @@ int motor_ramp_thread_main(int argc, char *argv[])
 {
 	_thread_running = true;
 
-	char *dev = PWM_OUTPUT0_DEVICE_PATH;
 	unsigned long max_channels = 0;
+	static struct pwm_output_values last_spos;
+	static struct pwm_output_values last_min;
+	static unsigned servo_count;
 
-	int fd = px4_open(dev, 0);
+	int fd = px4_open(_pwm_output_dev, 0);
 
 	if (fd < 0) {
-		PX4_ERR("can't open %s", dev);
+		PX4_ERR("can't open %s", _pwm_output_dev);
```

Comment:
Missing a `return 1`
Also `_thread_running` needs to be set to false on each return.